### PR TITLE
feat(acl): path templates + computed rules

### PR DESCRIPTION
**Pull request by yashgorana**
This pull request refactors and improves the ACL (Access Control List) system, focusing on code clarity, naming consistency, and efficiency. The main changes include renaming constants for clarity, updating function signatures and usages for more robust access checking, improving cache and rule handling, and cleaning up related tests to match the new API.

**ACL Constant and Naming Improvements:**
* Renamed constants in `internal/aclspec/aclspec.go` for clarity (e.g., `ACLFileName` → `FileName`, `Everyone` → `TokenEveryone`, `SetTerminal`/`UnsetTerminal` → `Terminal`/`NotTerminal`) and updated all usages throughout the codebase and tests. [[1]](diffhunk://#diff-91bc5d284cc060af32df9e79f150318c57c366b46514bbe9de11a771d4af4accL10-R35) [[2]](diffhunk://#diff-b9bc16988400e4b4200d581d5d8457b63e014a82a9cc3a8bd2288f637ef33376L86-R86) [[3]](diffhunk://#diff-b9bc16988400e4b4200d581d5d8457b63e014a82a9cc3a8bd2288f637ef33376L96-R96) [[4]](diffhunk://#diff-b9bc16988400e4b4200d581d5d8457b63e014a82a9cc3a8bd2288f637ef33376L110-R110) [[5]](diffhunk://#diff-439b44193e8665e925bab2e8079dd0825f7e5eb8d7b87ef7d65ac70d5d2fa0e2L137-R137) [[6]](diffhunk://#diff-439b44193e8665e925bab2e8079dd0825f7e5eb8d7b87ef7d65ac70d5d2fa0e2L149-R149) [[7]](diffhunk://#diff-c59c8788ed57654b4830a9faca5e47e77bb5ffb955bdb530d2b133a6910111e4L188-R188) [[8]](diffhunk://#diff-c59c8788ed57654b4830a9faca5e47e77bb5ffb955bdb530d2b133a6910111e4L197-R209) [[9]](diffhunk://#diff-c59c8788ed57654b4830a9faca5e47e77bb5ffb955bdb530d2b133a6910111e4L223-R224) [[10]](diffhunk://#diff-c59c8788ed57654b4830a9faca5e47e77bb5ffb955bdb530d2b133a6910111e4L242-R243) [[11]](diffhunk://#diff-c59c8788ed57654b4830a9faca5e47e77bb5ffb955bdb530d2b133a6910111e4L255-R259) [[12]](diffhunk://#diff-c59c8788ed57654b4830a9faca5e47e77bb5ffb955bdb530d2b133a6910111e4L275-R274) [[13]](diffhunk://#diff-c59c8788ed57654b4830a9faca5e47e77bb5ffb955bdb530d2b133a6910111e4L326-R332) [[14]](diffhunk://#diff-c59c8788ed57654b4830a9faca5e47e77bb5ffb955bdb530d2b133a6910111e4L339-R358) [[15]](diffhunk://#diff-c59c8788ed57654b4830a9faca5e47e77bb5ffb955bdb530d2b133a6910111e4L359-R374) [[16]](diffhunk://#diff-c59c8788ed57654b4830a9faca5e47e77bb5ffb955bdb530d2b133a6910111e4L380-R396)

**ACL Access Checking and API Refactor:**
* Refactored the `CanAccess` method in `internal/server/acl/acl.go` to use a new `ACLRequest` struct, improving clarity and extensibility. The method now checks the cache more efficiently, updates the cache on access decisions, and uses the new rule compilation logic.
* Removed the `GetRule` method from `ACLService`, consolidating rule lookup into the tree's `GetCompiledRule` method and updating all test cases to use the new API. [[1]](diffhunk://#diff-ae97a4ab0b490c0919c78b5686bf80c21dd3433b3316ee403d60b162f7760465L107-R151) [[2]](diffhunk://#diff-c59c8788ed57654b4830a9faca5e47e77bb5ffb955bdb530d2b133a6910111e4L197-R209) [[3]](diffhunk://#diff-c59c8788ed57654b4830a9faca5e47e77bb5ffb955bdb530d2b133a6910111e4L242-R243) [[4]](diffhunk://#diff-c59c8788ed57654b4830a9faca5e47e77bb5ffb955bdb530d2b133a6910111e4L255-R259) [[5]](diffhunk://#diff-c59c8788ed57654b4830a9faca5e47e77bb5ffb955bdb530d2b133a6910111e4L275-R274) [[6]](diffhunk://#diff-c59c8788ed57654b4830a9faca5e47e77bb5ffb955bdb530d2b133a6910111e4L326-R332) [[7]](diffhunk://#diff-c59c8788ed57654b4830a9faca5e47e77bb5ffb955bdb530d2b133a6910111e4L339-R358) [[8]](diffhunk://#diff-c59c8788ed57654b4830a9faca5e47e77bb5ffb955bdb530d2b133a6910111e4L380-R396)

**Cache and Rule Management Enhancements:**
* Improved cache management and debug logging for rule set updates and removals, ensuring accurate reporting of cache state and deletions. [[1]](diffhunk://#diff-ae97a4ab0b490c0919c78b5686bf80c21dd3433b3316ee403d60b162f7760465L96-R87) [[2]](diffhunk://#diff-ae97a4ab0b490c0919c78b5686bf80c21dd3433b3316ee403d60b162f7760465L107-R151) [[3]](diffhunk://#diff-ae97a4ab0b490c0919c78b5686bf80c21dd3433b3316ee403d60b162f7760465L226-L238)
* Adjusted blob change handling to only remove cache entries for non-ACL files and improved debug output for cache removals.

**Test and Dependency Updates:**
* Updated all affected tests to use the new constants and the refactored access checking/request API, ensuring coverage of the new code paths. [[1]](diffhunk://#diff-c59c8788ed57654b4830a9faca5e47e77bb5ffb955bdb530d2b133a6910111e4L197-R209) [[2]](diffhunk://#diff-c59c8788ed57654b4830a9faca5e47e77bb5ffb955bdb530d2b133a6910111e4L242-R243) [[3]](diffhunk://#diff-c59c8788ed57654b4830a9faca5e47e77bb5ffb955bdb530d2b133a6910111e4L255-R259) [[4]](diffhunk://#diff-c59c8788ed57654b4830a9faca5e47e77bb5ffb955bdb530d2b133a6910111e4L275-R274) [[5]](diffhunk://#diff-c59c8788ed57654b4830a9faca5e47e77bb5ffb955bdb530d2b133a6910111e4L326-R332) [[6]](diffhunk://#diff-c59c8788ed57654b4830a9faca5e47e77bb5ffb955bdb530d2b133a6910111e4L339-R358) [[7]](diffhunk://#diff-c59c8788ed57654b4830a9faca5e47e77bb5ffb955bdb530d2b133a6910111e4L380-R396)
* Bumped `github.com/bmatcuk/doublestar/v4` dependency from v4.8.1 to v4.9.1 in `go.mod`.

**Minor Cleanups:**
* Removed unused imports and dead code from `internal/server/acl/acl.go` for better maintainability. [[1]](diffhunk://#diff-ae97a4ab0b490c0919c78b5686bf80c21dd3433b3316ee403d60b162f7760465L5-L8) [[2]](diffhunk://#diff-ae97a4ab0b490c0919c78b5686bf80c21dd3433b3316ee403d60b162f7760465L226-L238)

These changes collectively make the ACL system more maintainable, robust, and easier to extend in the future.

### CODE REVIEW

File: internal/server/acl/utils.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+			}
+		case '?', '!', '[', '{':
+			score -= 2 // Non * wildcards get smaller penalty
+		}
+	}
+
+	return score
+}
+
+func sortRulesBySpecificity(rules []*aclspec.Rule) []*aclspec.Rule {
```
Comment (madhavajay) [inline internal/server/acl/utils.go:45]: We probably need more tests for this in case of weird edge cases, sort order and tie breaks etc.

Comment (shubham3121) [reply internal/server/acl/utils.go:45]: Added tests for this [a5714d1](https://github.com/OpenMined/syftbox/pull/62/commits/a5714d1e444c2e7a45a89c1cbfff402a7215f4ca)
